### PR TITLE
Add format check and format code

### PR DESCRIPTION
**Pull request by ericphanson**

### CODE REVIEW

**Comment by ericphanson** [comment]
CI is failing because of https://github.com/domluna/JuliaFormatter.jl/issues/343, I'll work around it here by using `NamedTuple()` instead

**Comment by ericphanson** [comment]
Before I fix it, I'll run CI again to get the failing badge (edit: for testing something, turned out to be a misconfig)

**Comment by ericphanson** [comment]
This seemed to be failing even though I had formatted locally without any changes. Going to leave this for now...

**Comment by ericphanson** [comment]
> This seemed to be failing even though I had formatted locally without any changes

I had formatted with YASStyle but did not choose the style for the format check! Resolved it by making it use the same Manifest and `run.jl` script (which of course I should've done from the start). This should be fairly simple and robust now I think, since the check just calls `run.jl` in the `format` project on Julia 1.5.3 and checks if any files have changed, which is exactly how I'd do the check locally.

**Comment by hannahilea** [comment]
Just to clarify, this covers "fail if linting fails" but doesn't yet do anything fancy like "open a PR with the lint fix", right?

File: .github/workflows/format_check.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+name: format-check
+
+on:
+  push:
+    branches:
+      - 'master'
```
Comment (hannahilea) [inline .github/workflows/format_check.yml]: should this be `main`?

File: .github/workflows/format_check.yml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+      - 'release-'
```
Comment (hannahilea) [inline .github/workflows/format_check.yml]: potentially not relevant to our workflows?

**Review by hannahilea — APPROVED** [review]
LGTM barring potential branch clarification

**Comment by ericphanson** [comment]
> Just to clarify, this covers "fail if linting fails" but doesn't yet do anything fancy like "open a PR with the lint fix", right?

Yeah, exactly. I saw a workflow to open PRs, https://github.com/julia-actions/julia-format/blob/master/workflows/format_pr.yml, but I think it opens them against `main` (or master maybe), but I'd rather it opens them against non-merged PRs that fail the format check. Probably that's possible somehow, but I'm not really sure how.

File: .github/workflows/format_check.yml (hunk 2 above)
Comment (ericphanson) [reply .github/workflows/format_check.yml]: yeah, I'm not sure. I've switched it to the regex that's in `ci.yml` so at least it's consistent across workflows. I've used release branches with Convex.jl once or twice when I want to backport a fix to a previous version, so I think it's reasonable to at least have CI that understands that use-case, even if it's rare.

**Pull request merged**